routes/drones.js

```diff
@@ -1,36 +1,71 @@
 const express = require('express');
 const router = express.Router();
-
+const Drones = require(`../models/drone.model`);
 // require the Drone model here
 
-router.get('/drones', (req, res, next) => {
-  // Iteration #2: List the drones
-  // ... your code here
-});
+router.get('/drones', async(req, res, next) => {
+  try{
+  let drones = await Drones.find();
+        res.render (`drones/list`,{drones});
 
-router.get('/drones/create', (req, res, next) => {
-  // Iteration #3: Add a new drone
-  // ... your code here
+      }catch(error){
+  next(error)
+      }
 });
 
-router.post('/drones/create', (req, res, next) => {
-  // Iteration #3: Add a new drone
-  // ... your code here
+router.get(`/drones/create`, (req, res) => {res.render(`drones/create-form`)});
+
+router.post('/drones/create', async (req, res, next) => {
+  try{
+    const {name, propellers, speed} = req.body
+    await Drones.create({name, propellers, speed})
+    res.redirect(`/drones`);
+  } catch (error){
+    console.log (error);
+    next(error);
+  }
 });
 
-router.get('/drones/:id/edit', (req, res, next) => {
-  // Iteration #4: Update the drone
-  // ... your code here
+
+
+router.get('/drones/:id/edit', async (req, res, next) => {
+  try{
+    const {id} = req.params;
+    const book = await Drone.findById (id);
+    res.render (`drones/update-form`, drone);
+
+} catch (error){
+    console.log(error);
+    next(error);
+}
 });
 
-router.post('/drones/:id/edit', (req, res, next) => {
-  // Iteration #4: Update the drone
-  // ... your code here
+router.post('/drones/:id/edit', async(req, res, next) => {
+  try{
+    const {id} = req.params;
+    const {name, propellers, speed} = req.body ;
+    
+    await Drone.findByIdAndUpdate(id, {name, propellers, speed});
+
+
+       res.redirect (`/drones/${id}`);
```

Comment:
Here, this isn't working because the route ``(`/drones/${id}`)`` doesn't exist. Please always check what routes you have before doing anything for your organisation. So here, you have various options, but not the one you passed. You could've redirected to the home page `('/')` or back to drones `('/drones')`